crypto/address/address.go

```diff
@@ -0,0 +1,298 @@
+package address
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"github.com/orbs-network/orbs-network-go/crypto/base58"
+	"golang.org/x/crypto/ripemd160"
+	"hash/crc32"
+)
+
+type Address struct {
+	networkId      string
+	virtualChainId []byte
+	publicKey      []byte
+	version        uint8
+	accountId      []byte
+	checksum       *uint32
+	fullAddress    []byte
+}
+
+const (
+	MAIN_NETWORK_ID      = "M"
+	TEST_NETWORK_ID      = "T"
+	ADDRESS_LENGTH       = 40
+	VIRTUAL_CHAIN_ID_MSB = 0X08
+	VCHAIN_ID_SIZE       = 3
+	NETWORK_ID_SIZE      = 1
+	VERSION_SIZE         = 1
+	ACCOUNT_ID_SIZE      = 20
+	CHECKSUM_SIZE        = 4
+	PUBLIC_KEY_SIZE      = 32
+)
+
+func validateAddressParts(pk []byte, vcId []byte, net string, v uint8) error {
+	if len(pk) != PUBLIC_KEY_SIZE {
+		return fmt.Errorf("pk invalid, cannot create address")
```

Comment:
we need to be consistent with regards to error types. So far, @jlevison is for using `Errorf` while @talkol is for using concrete types.

Regardless, please add the offending field to the error string so that caller knows what they sent wrong.

Reply:
added errors.Wrap where needed